components/engine_rocks/src/raft_compaction_filter.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::raft_engine::RAFT_LOG_GC_INDEXES;
+use crate::raw::{
+    new_compaction_filter_raw, CompactionFilter, CompactionFilterContext, CompactionFilterDecision,
+    CompactionFilterFactory, CompactionFilterValueType, DBCompactionFilter,
+};
+use std::{collections::HashMap, ffi::CString};
+use tikv_util::debug;
+
+pub struct RaftLogCompactionFilterFactory {}
+
+impl RaftLogCompactionFilterFactory {}
+
+impl CompactionFilterFactory for RaftLogCompactionFilterFactory {
+    fn create_compaction_filter(
+        &self,
+        _context: &CompactionFilterContext,
+    ) -> *mut DBCompactionFilter {
+        let filter = Box::new(RaftLogCompactionFilter::new());
+        let name = CString::new("raft_log_gc_compaction_filter").unwrap();
+        unsafe { new_compaction_filter_raw(name, filter) }
+    }
+}
+
+struct RaftLogCompactionFilter {
+    map: HashMap<u64, u64>,
```

Comment:
Regions are met one by one, they are sorted. So use a `current_region_id` is enough.

Reply:
Get it!